src/tbp/monty/frameworks/models/evidence_matching/resampling_hypotheses_updater.py

```diff
@@ -222,6 +248,31 @@ def __init__(
         # Dictionary of resampling telemetry for each channel in each graph_id
         self.resampling_telemetry: dict[str, dict[str, HypothesesUpdateTelemetry]] = {}
 
+        # Trigger a burst at the beginning of the episode
+        self.sampling_burst_steps = self.sampling_burst_duration
+
+    def pre_step(self) -> None:
+        """Runs once per step before updating the hypotheses.
+
+        We calculate the max slope and update resampling parameters before running the
+        hypotheses update loop/threads over all the graph_ids and channels.
+        """
+        self.max_slope = self._max_global_slope()
+
+        if (
+            self.max_slope <= self.slope_burst_trigger
+            and self.sampling_burst_steps == 0
+        ):
+            self.sampling_burst_steps = self.sampling_burst_duration
+
+    def post_step(self) -> None:
```

Comment:
Do we need to do this in a separate post_step method? Couldn't we just decremen it in the `if self.sampling_burst_steps > 0:` branch in the `sample_count` function?